src/main/java/dk/bec/polonez/reservationsystem/controller/OfferController.java

```diff
@@ -20,18 +20,32 @@ public OfferController(OfferService offerService) {
     }
 
     @GetMapping
-    public List<Offer> getAllOffers() {
+    public List<OfferDto> getAllOffers() {
         return offerService.getAll();
     }
 
     @GetMapping("{id}")
-    public Offer getOfferById(@PathVariable long id) {
+    public OfferDto getOfferById(@PathVariable long id) {
         return offerService.getById(id);
     }
 
     @PostMapping
-    @ResponseStatus(HttpStatus.CREATED)
-    public ResponseOfferDto addOffer(@RequestBody CreateOfferDto offerDto) {
+    public OfferDto addOffer(@RequestBody CreateOfferDto offerDto) {
         return offerService.addOffer(offerDto);
     }
+
+    @PutMapping("{id}")
+    public OfferDto updateOffer(@PathVariable long id, @RequestBody CreateOfferDto offerDto) {
```

Comment:
**advice** No validation of request body, could be added in this or subsequent pull-requets.

Reply:
You meant validation within annotations for instance "@Valid"? I might do something like in this [tutorial](https://www.baeldung.com/spring-boot-bean-validation) also with exceptions returning particular messages regarding invalid field.

Reply:
Yes, exactly.